### DISCUSSION TITLE
¿Is there an endpoint to get the data from the transaction view??

### DISCUSSION THREAD

**Discussion by felduque557** [amzn/selling-partner-api-models#3850] [General] [answered]
I currently have a public developer profile and I need to manage transactions for each store, service fee, Refund, etc. But I can't find any endpoint to provide me with this information, could someone help me?
![Screenshot_2](https://github.com/amzn/selling-partner-api-docs/assets/134432610/bf486d33-f3b8-4465-b624-da847fedbbc5)

**Comment by rugved1991** [comment, marked as answer]
Hi @felduque557,

Currently it is not possible to create a custom date range transaction report through the API. But, you can create a report on the UI and then access it through the [getReports](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-reference#getreports) operation of Reports API.
You can also take a look at the [Finance API](https://developer-docs.amazon.com/sp-api/docs/finances-api-reference) to get settlement information through the API.
 
Best,
Rugved
Solutions Architect, SP-API

**Reply by felduque557** [reply]
Hi @rugved1991 

Do you know how can I generate the reports from the API and then get all the information or with what reportype

![Screenshot_3](https://github.com/amzn/selling-partner-api-docs/assets/134432610/eecba25e-48ab-4132-ae47-113340634db1)

**Reply by yohaiai** [reply]
This is very helpful, i was looking for the same, but for Summary report. Do you know how to get the income, expenses and transactions totals through the API/Finance API?

![Uploading Screen Shot 2024-04-24 at 16.15.48.png…]()

**Reply by felduque557** [reply]
@yohairosen I am currently using is the reporting api to get that information only some reportType says you have no permissions or they are temporarily disabled.

https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-reference

**Reply by yohaiai** [reply]
Thank you @felduque557 , can you share the specific report type you are using?

**Reply by felduque557** [reply]
This type of report is the closest I have found to delivering amazon fees, removals, refunds and orders. I hope you find it useful, I had a hard time finding it. 

https://sellingpartnerapi-na.amazon.com/reports/2021-06-30/reports?reportTypes=GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_V2&marketplaceIds=ATVPDKIKX0DER&createdSince=2024-02-01T00:00:00

Reportype:  GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_V2

Note: This type of report cannot be created, nor can schemas be created, it is generated automatically by amazon with a 90 day history, divided into 14 day reports.

**Reply by np-data** [reply]
Hi @felduque557 
I believe you are right here. GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_V2 is the closest to manual transaction report under payments reports repository.

But I would like to highlight that the report generated using API does not:
1) separate standard order and invoice order data (I believe invoiced orders have a null transaction type).
2) For any given week, I'm seeing very less data in the API report than manual.

Would appreciate your response if you're observing the same differences.

**Reply by felduque557** [reply]
> Hi @felduque557 I believe you are right here. GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_V2 is the closest to manual transaction report under payments reports repository.
> 
> But I would like to highlight that the report generated using API does not:
> 
> 1. separate standard order and invoice order data (I believe invoiced orders have a null transaction type).
> 2. For any given week, I'm seeing very less data in the API report than manual.
> 
> Would appreciate your response if you're observing the same differences.

Hi @np-data 
On the first point I really agree, standard and invoiced transactions go together. It is really complicated to manage the invoiced ones, so it is actually money that the store has not yet received in some cases. Even more limiting is that we can only access a 90 day history. 

Regarding the second point for now I have not encountered that problem since I only store the transactions and then make use of it.

It would be nice if at some point the amazon team would allow us to generate the report by date range so we could have the information up to date and not wait 14 days to update it. As well as the report of refunds and returns.

**Reply by np-data** [reply]
That makes sense. Thanks!
